src/PrizeVault.sol

```diff
@@ -0,0 +1,750 @@
+    function maxDeposit(address) public view returns (uint256) {
+        uint256 _totalSupply = totalSupply();
+        if (totalAssets() < _totalSupply) return 0;
+
+        // the vault will never mint more than 1 share per asset, so no need to convert supply buffer to assets
+        uint256 _twabSupplyLimit = type(uint96).max - _totalSupply;
+        uint256 _maxDeposit;
+        uint256 _latentBalance = _asset.balanceOf(address(this));
+        uint256 _maxYieldVaultDeposit = yieldVault.maxDeposit(address(this));
+        if (_latentBalance >= _maxYieldVaultDeposit) {
+            return 0;
+        } else {
+            unchecked {
+                _maxDeposit = _maxYieldVaultDeposit - _latentBalance;
+            }
+            return _twabSupplyLimit < _maxDeposit ? _twabSupplyLimit : _maxDeposit;
+        }
+    }
+
+    /// @inheritdoc IERC4626
+    /// @dev Returns the same value as `maxDeposit` since shares and assets are 1:1 on mint.
+    function maxMint(address _owner) public view returns (uint256) {
+        return maxDeposit(_owner);
+    }
+
+    /// @inheritdoc IERC4626
+    /// @dev TODO: add reasoning for inclusion of latent balance
+    function maxWithdraw(address _owner) public view returns (uint256) {
+        uint256 _maxWithdraw = yieldVault.maxWithdraw(address(this)) + _asset.balanceOf(address(this));
+
+        // the owner may receive less than 1 asset per share, so we must convert their balance here
+        uint256 _ownerAssets = convertToAssets(balanceOf(_owner));
+        return _ownerAssets < _maxWithdraw ? _ownerAssets : _maxWithdraw;
+    }
+
+    /// @inheritdoc IERC4626
+    /// @dev TODO: add reasoning for inclusion of latent balance
+    function maxRedeem(address _owner) public view returns (uint256) {
+        // the owner will never receive more than 1 asset per share, so no need to convert max withdraw to shares
+        uint256 _maxRedeem = yieldVault.maxWithdraw(address(this)) + _asset.balanceOf(address(this));
+        uint256 _ownerShares = balanceOf(_owner);
+        return _ownerShares < _maxRedeem ? _ownerShares : _maxRedeem;
+    }
+
+    /// @inheritdoc IERC4626
+    function previewDeposit(uint256 _assets) public pure returns (uint256) {
+        // shares represent how many assets an account has deposited, so they are 1:1 on deposit
+        return _assets;
+    }
+
+    /// @inheritdoc IERC4626
+    function previewMint(uint256 _shares) public pure returns (uint256) {
+        // shares represent how many assets an account has deposited, so they are 1:1 on mint
+        return _shares;
+    }
+
+    /// @inheritdoc IERC4626
+    /// @dev Reverts if `totalAssets` in the vault is zero
+    function previewWithdraw(uint256 _assets) public view returns (uint256) {
+        uint256 _totalAssets = totalAssets();
+
+        // No withdrawals can occur if the vault controls no assets.
+        if (_totalAssets == 0) revert ZeroTotalAssets();
```

Comment:
`previewWIthdraw` returns the shares that will be burned in order to withdraw the assets. If we return zero, it would be saying that if the user withdrew 'x' assets, it would cost them zero shares, which is inaccurate.

Reply:
That makes sense!